test_treematcher.py

```diff
@@ -116,96 +120,78 @@ def test_lineages(self):
+        t9 = PhyloTree("(9305.ENSSHAP00000009662:1,9305.ENSSHAP00000009620:1);")
+        t10 = PhyloTree("((9315.ENSMEUP00000008285:0.899711,9258.ENSOANP00000027752:0.559777)0.99985:0.11989,((9739.ENSTTRP00000010720:0.164873,9913.ENSBTAP00000003500:0.298158)0.99985:0.109903,((9685.ENSFCAP00000006440:0.239731,(9615.ENSCAFP00000042310:0.122399,(9646.ENSAMEP00000002314:0.18278,9669.ENSMPUP00000005544:0.270727)0.6117:0.0396991)0.99985:0.0702148)0.99985:0.082488,(132908.ENSPVAP00000014833:0.488081,(9796.ENSECAP00000022144:0.310699,(((9785.ENSLAFP00000009512:0.187095,9813.ENSPCAP00000004417:0.493329)0.99985:0.359095,(30611.ENSOGAP00000016876:0.334272,(9483.ENSCJAP00000021314:0.178043,(9601.ENSPPYP00000003401:0.0415077,((61853.ENSNLEP00000003253:0.196659,9544.ENSMMUP00000037769:0.326984)0.835225:0.0989423,(9593.ENSGGOP00000004740:0.101826,9606.ENSP00000182290:0.0204981)0.997196:0.020731)0.307827:0.0046059)0.99985:0.0991112)0.99985:0.162323)0.972253:0.0380139)0.70642:0.0193389,((10141.ENSCPOP00000016274:0.272126,43179.ENSSTOP00000015376:0.458416)0.996119:0.0901785,(37347.ENSTBEP00000013312:0.328061,(10020.ENSDORP00000010739:0.398341,(10116.ENSRNOP00000051746:0.0455948,10090.ENSMUSP00000009396:0.0811741)0.99985:0.269525)0.791467:0.0577236)0.536676:0.0461933)0.99985:0.0620583)0.99985:0.0788824)0.969465:0.0395994)0.635969:0.0171601)0.702925:0.0283261)0.99985:0.11989);")
+
+        trees = [(t1, "t1", True), (t2, "t2", False), (t3, "t3", True),
+                 (t4, "t4", True), (t5, "t5", True), (t6, "t6", False),
+                 (t7, "t7", True), (t8, "t8", True), (t9, "t9", False),
+                 (t10, "t10", True)]
+        for tree, tree_name, has_matches in trees:
             tree.set_species_naming_function(lambda n: n.name.split(".")[0] if "." in n.name else '')
             tree.annotate_ncbi_taxa()
-
-
             pattern = """
                 ( ' 9443 in @.lineage ' , ' 9443 in @.lineage and @.name!=9606 ' )' @.support >= 0.9 ';
                 """
-            pattern = TreePattern(pattern, format=8, quoted_node_names=False)
-
-            self.assertEqual(list(pattern.find_match(tree, None))[0].lineage, lineages[index])
-
-            index += 1
+            pattern = TreePattern(pattern)
+            if not has_matches:
+                self.assertEqual(list(pattern.find_match(tree)), [])
+            else:
+                match = pattern.find_match(tree).next()
```

Comment:
Check if the match is correct by manually checking every condition requested in the pattern